typings/externs.d.ts

```diff
@@ -4,139 +4,141 @@
-  debugString?: string;
-  score?: number;
-  optimalValue: number | string;
-  extendedInfo?: {value: string};
-}
-
-export interface AuditResults {
-  [metric: string]: AuditResult;
-}
-
-export interface AuditFullResult {
-  rawValue: boolean | number;
-  displayValue: string;
-  debugString?: string;
-  score: number;
-  scoreDisplayMode: string;
-  error?: boolean;
-  description: string;
-  name: string;
-  helpText?: string;
-  extendedInfo?: {value: string};
-}
-
-export interface AuditFullResults {
-  [metric: string]: AuditFullResult;
-}
-
-export interface Results {
-  url: string;
-  audits: AuditFullResults;
-  lighthouseVersion: string;
-  artifacts?: Object;
-  initialUrl: string;
-  generatedTime: string;
-}
-
-export interface LaunchedChrome {
-  pid: number;
-  port: number;
-  kill: () => Promise<{}>;
-}
-
-export interface LighthouseError extends Error {
-  code?: string;
-  friendlyMessage?: string;
-}
-
-export interface TraceEvent {
-  name: string;
-  args: any;
-  tid: number;
-  ts: number;
-  dur: number;
-}
-
-export interface NetworkRequest {
-  requestId: string;
-  connectionId: string;
-  connectionReused: boolean;
-
-  url: string;
-  protocol: string;
-  origin: string | null;
-  parsedURL: DevToolsParsedURL;
-
-  startTime: number;
-  endTime: number;
-
-  transferSize: number;
-
-  _initiator: NetworkRequestInitiator;
-  _timing: NetworkRequestTiming;
-  _resourceType: any;
-  priority(): 'VeryHigh' | 'High' | 'Medium' | 'Low';
-}
-
-export interface NetworkRequestInitiator {
-  type: 'script' | 'parser';
-}
-
-export interface NetworkRequestTiming {
-  connectStart: number;
-  connectEnd: number
-  sslStart: number;
-  sslEnd: number;
-  sendStart: number;
-  sendEnd: number;
-  receiveHeadersEnd: number;
-}
-
-export interface DevToolsParsedURL {
-  scheme: string;
-  host: string;
-}
-
-export interface DevToolsJsonTarget {
-  description: string;
-  devtoolsFrontendUrl: string;
-  id: string;
-  title: string;
-  type: string;
-  url: string;
-  webSocketDebuggerUrl: string;
+declare namespace LH {
+
+  export interface Flags {
+    _: string[];
+    port: number;
+    chromeFlags: string;
+    output: any;
+    outputPath: string;
+    saveAssets: boolean;
+    view: boolean;
+    maxWaitForLoad: number;
+    logLevel: string;
+    hostname: string;
+    blockedUrlPatterns: string[];
+    extraHeaders: string;
+    enableErrorReporting: boolean;
+    listAllAudits: boolean;
+    listTraceCategories: boolean;
+    auditMode: boolean;
+    gatherMode: boolean;
+    configPath?: string;
+    perf: boolean;
+    mixedContent: boolean;
+    verbose: boolean;
+    quiet: boolean;
+  }
+
+  export interface Config {}
+
+  export interface AuditResult {
+    rawValue: boolean | number | null;
+    displayValue?: string;
+    debugString?: string;
+    score?: number;
+    extendedInfo?: {value: string};
+    notApplicable?: boolean;
+    error?: boolean;
+    // TODO: define details
+    details?: object;
```

Comment:
(for when it's time to take on `details`...) 

i did an initial spike of `details`'s types in https://github.com/GoogleChrome/lighthouse/compare/95e19a56503c702a5077c62dda1112cd39c4d705...typechecking#diff-518173330fe66b5974ceeb9a14a7b215


But then https://docs.google.com/document/d/1TumV5t3xtJ_e2b8y9PTwh4j2RaT-lmaQI9lwmF5tD2U/edit#heading=h.y9rlfi2u13u9 has a more future-forward definition, which is even stricter.